### PR TITLE
Force installation of stable version of phpstan

### PR DESCRIPTION
**Pull request by maxime-rainville**
Fix PHP-Stan to release 0.9 which the latest stable release.

All unit tests are green when using this version.

# Parent issue
https://github.com/silverstripe/silverstripe-upgrader/issues/70

### CODE REVIEW

**Comment by dhensby** [comment]
@tractorcow any idea why you went with 0.10 in this commit (https://github.com/silverstripe/silverstripe-upgrader/commit/c75b82872c618e4468b8cffdd6cadf9b0e86d1df) ?

**Comment by maxime-rainville** [comment]
@dhensby I had a chat with @tractorcow about that before he buggered off on holiday. He couldn't recall exactly why he did it. It didn't seem like a big deal. But we didn't discuss it in detail.

The alternative to requiring `^0.9` would be to require a specific commit that we know works.

Or I could update our code base to work with the new API for PhpStan. However, it seems like the maintainer of that package is introducing breaking changes to this dev branch, so there's no guarantee they won't be more API change later on.

If we want to be extra sure `^0.9` is functionally equivalent for our purposes, I can have a thorough look at the unit tests for classes where we use PHPStan and beef them up a bit.

**Comment by dhensby** [comment]
OK - if we are seeing breaking changes in the 0.10 line (which is allowed in the 0.* lines) then we should stick to `~0.9.0` instead of `^0.9` as `^0.9` will pull in 0.10 when it's stable.

File: composer.json [hunk 1]
```diff
@@ -14,8 +14,8 @@
         "symfony/console": "^3.2",
         "symfony/yaml": "^3.2",
         "phpspec/php-diff": "^1",
-        "phpstan/phpstan": "^0.10",
-        "phpstan/phpstan-php-parser": "^0.10",
+        "phpstan/phpstan": "^0.9.0",
```
Comment (dhensby) [inline composer.json]:
```diff
-"phpstan/phpstan": "^0.9.0",
+"phpstan/phpstan": "~0.9.0",

Comment (maxime-rainville) [reply composer.json]: Good point. I'll fix `phpstan/phpstan-php-parser` and `phpstan/phpstan` to `~0.9.0`.

Let's revisit this when phpstan reaches `1.0.0`

**Comment by unclecheese** [comment]
Closing https://github.com/silverstripe/silverstripe-upgrader/issues/70 as this has been merged.

**Pull request merged**